miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/ExperimentRestController.java

```diff
@@ -26,23 +26,126 @@
 import java.io.IOException;
 import java.util.Collection;
 
+import javax.ws.rs.core.Response.Status;
+
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
 import org.springframework.web.bind.annotation.ResponseBody;
 
+import uk.ac.bbsrc.tgac.miso.core.data.ChangeLog;
 import uk.ac.bbsrc.tgac.miso.core.data.Experiment;
+import uk.ac.bbsrc.tgac.miso.core.data.Experiment.RunPartition;
+import uk.ac.bbsrc.tgac.miso.core.data.Kit;
+import uk.ac.bbsrc.tgac.miso.core.data.Partition;
+import uk.ac.bbsrc.tgac.miso.core.data.Run;
+import uk.ac.bbsrc.tgac.miso.dto.Dtos;
+import uk.ac.bbsrc.tgac.miso.dto.ExperimentDto;
+import uk.ac.bbsrc.tgac.miso.dto.KitConsumableDto;
+import uk.ac.bbsrc.tgac.miso.service.ChangeLogService;
+import uk.ac.bbsrc.tgac.miso.service.ContainerService;
 import uk.ac.bbsrc.tgac.miso.service.ExperimentService;
+import uk.ac.bbsrc.tgac.miso.service.KitService;
+import uk.ac.bbsrc.tgac.miso.service.impl.RunService;
 
 @Controller
 public class ExperimentRestController extends RestController {
 
   @Autowired
+  private ChangeLogService changeLogService;
+  @Autowired
+  private ContainerService containerService;
+  @Autowired
   private ExperimentService experimentService;
+  @Autowired
+  private KitService kitService;
+  @Autowired
+  private RunService runService;
 
-  @RequestMapping(value = "/experiments/rest/", method = RequestMethod.GET)
+  @RequestMapping(value = "/rest/experiment/{experimentId}/addkit", method = RequestMethod.POST, produces = "application/json")
+  public @ResponseBody KitConsumableDto addKit(@PathVariable Long experimentId, @RequestBody KitConsumableDto dto) throws IOException {
+    Experiment experiment = experimentService.get(experimentId);
+    if (experiment == null) {
+      throw new RestException("No such experiment.", Status.NOT_FOUND);
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "No such experiment." 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)